src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlAnyElementAttributes.cs

```diff
@@ -144,43 +165,67 @@ void ICollection.CopyTo(Array array, int index)
         {
             get
             {
-                return InnerList[index];
+                return List[index];
             }
             set
             {
-                InnerList[index] = value;
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                List[index] = value;
             }
         }
 
         bool IList.Contains(Object value)
         {
-            return InnerList.Contains(value);
+            return List.Contains(value);
         }
 
         int IList.Add(Object value)
         {
+            if (value == null)
+            {
+                throw new ArgumentException(nameof(value));
+            }
+
             return List.Add(value);
         }
 
         void IList.Remove(Object value)
         {
-            int index = InnerList.IndexOf(value);
-            InnerList.RemoveAt(index);
+            if (value == null)
+            {
+                throw new ArgumentException(nameof(value));
+            }
+
+            var attribute = value as XmlAnyElementAttribute;
+            if (attribute == null)
+            {
+                throw new ArgumentException(SR.Arg_RemoveArgNotFound);
+            }
+            Remove(attribute);
         }
 
         int IList.IndexOf(Object value)
         {
-            return InnerList.IndexOf(value);
+            return List.IndexOf(value);
         }
 
         void IList.Insert(int index, Object value)
         {
-            InnerList.Insert(index, value);
+            if (value == null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+
+            List.Insert(index, value);
         }
 
-        public IEnumerator GetEnumerator()
+        IEnumerator IEnumerable.GetEnumerator()
```

Comment:
In X509CertificateCollection.cs as mentioned above, we have very similar issue regarding the difference between Desktop and CoreClr. Turns out we don't use explicit implementation on Desktop so I'll revert this part. Nonetheless, without this part this PR is a breaking change.